### ISSUE TITLE
[FR] Mermaid diagram support

### ISSUE THREAD

**Issue by efemkay** [sissilab/obsidian-image-toolkit#71]
Mermaid diagram by default doesn't scale to the screen at all. If its big, it will simply overflow. I want to request feature to limit the diagram to screen size (scaled to width) and use the same toolkit to provide zoom on click similar to how it is for normal picture.

I note that this plugin does support svg (mermaid also output the diagram in svg) but it doesn't work currently. Perhaps related to #68

**Comment by sissilab** [comment]
@efemkay 

Hi, thanks for your feedback. I haven't used mermaid diagram yet. Did you mean that this plugin doesn't support svg when you write mermaid diagram code as follows. And the converted svg cannot be clicked to pop up.

```mermaid
graph TD
A --> B
B --> C
B --> D
```

**Comment by efemkay** [comment]
yeah. the svg converted from mermaid cannot be clicked. i think it's because even with svg as a source similar to jpg/png, it is wrapped with `<img>` tag. whereas mermaid diagram in obsidian it is rendered within an `<svg>` tag. i guess that helps -- that is as much as i can confidently say about it. if it helps, i'm able to scale it using css but i'm limited to `:active` pseudo selector (which means i need to hold the click to maintain the pseudo "zoom"

![image](https://user-images.githubusercontent.com/42369515/195835026-db4262a8-0a6b-4b4d-a9d1-d3400f6b4be9.png)
